src/fields/field.hh

```diff
@@ -368,6 +374,8 @@ protected:
 
     std::vector<std::shared_ptr<FactoryBase> >  factories_;
 
+    std::vector< typename Value::return_type > r_values_;
```

Comment:
We need cached values for elements of all dimensions. So r_values_ should possibly be a vector of vectors. Add documentation of r_values_.